src/Highlighter.js

```diff
@@ -66,9 +68,20 @@ export default function Highlighter ({
         if (chunk.highlight) {
           highlightCount++
 
+          if(textColors){
+            textColors = Object.keys(textColors).reduce((colors, x) => {
```

Comment:
Probably don't want to run this reduce every time we render. Seems like a good opportunity to use a lib like [memoize-one](https://www.npmjs.com/package/memoize-one) to avoid unnecessary work.

Reply:
That makes a lot of sense.